src/expr/src/relation/func.rs

```diff
@@ -580,10 +580,39 @@ fn lag_lead_inner_respect_nulls<'a>(
     result
 }
 
+#[allow(clippy::as_conversions)]
 fn lag_lead_inner_ignore_nulls<'a>(
     args: Vec<(Datum<'a>, Datum<'a>, Datum<'a>)>,
     lag_lead_type: &LagLeadType,
 ) -> Vec<Datum<'a>> {
+    if i64::try_from(args.len()).is_err() {
+        panic!("window partition way too big")
+    }
+    // Preparation: Make sure we can jump over a run of nulls in constant time, i.e., regardless of
+    // how many nulls the run has. The following skip tables will point to the next non-null index.
+    let mut skip_nulls_backward = Vec::new();
```

Comment:
The capacity could be specified upfront here

Reply:
Right!